tools/c7n_tencentcloud/c7n_tencentcloud/query.py

```diff
@@ -2,106 +2,270 @@
 
     def get_resource_query_params(self):
-        return {}
+        config_query = self.data.get("query", [])
+        if config_query:
+            return {
+                "Filters": config_query
+            }
+        else:
+            return {}
 
     def resources(self):
         params = self.get_resource_query_params()
         resources = self.source.resources(params)
-        return self.filter_resources(resources)
+
+        # filter resoures
+        resources = self.filter_resources(resources)
+
+        self.check_resource_limit(resources)
+        return resources
 
     def augment(self, resources):
         return resources
 
+    # TODO
+    # to support configs: max-resources, max-resources-percent
+    def check_resource_limit(self, resources):
+        return resources
+
 
-@sources.register("describe-tencentcloud")
+@sources.register(DESC_SOURCE_NAME)
 class DescribeSource:
-    """"""
+    """DescribeSource"""
     def __init__(self, resource_manager: QueryResourceManager) -> None:
+        """
+        :param query: The query to execute from query in policy.yaml
+        """
         self.resource_manager = resource_manager
-        self.query = ResourceQuery(resource_manager.session_factory)
+        self.resource_type = resource_manager.resource_type
+        self.region = resource_manager.config.region
+        self.query_helper = ResourceQuery(resource_manager.session_factory)
+        self._session = None
 
     def resources(self, params=None):
+        """
+        It returns a list of resources that match the given parameters
+
+        :param params: A dictionary of parameters to filter the list of resources returned
+        :return: A list of resources.
+        """
         if params is None:
             params = {}
-        return self.query.filter(self.resource_manager, params)
+
+        res = self.query_helper.filter(self.resource_manager.config.region,
+                                       self.resource_manager.resource_type,
+                                       params)
+        self.augment(res)
+        return res
 
     def get_permissions(self):
-        return None
+        return []
 
     def augment(self, resources):
+        for r in resources:
+            r["Tags"] = self.get_resource_tag(r)
         return resources
+
+    def get_resource_tag(self, resource):
+        """
+        Get resource tag
+        All resource tags need to be obtained separately
+        """
+        result = []
+        resouce_id = resource[self.resource_type.id]
+        ce6 = self.get_resource6(resouce_id)
+        res = self.query_helper.get_resource_tags(self.region, ce6)
+        if res:
+            # get the resource's tags
+            for tag in res[0]["Tags"]:
+                result.append({
+                    "Key": tag["TagKey"],
+                    "Value": tag["TagValue"]
+                })
+        return result
+
+    def get_resource6(self, resource_id):
```

Comment:
interesting, looks like arn by any other name. is there a another name for it, perhaps we call it a resource_descriptor  or `qcs`

Reply:
Yes, it is named qcs for tencentcloud as arn. Will rename it.